src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -416,16 +416,28 @@ public Document WithFilePath(string? filePath)
         => this.Project.Solution.WithDocumentFilePath(this.Id, filePath).GetRequiredDocument(Id);
 
     /// <summary>
-    /// Get the text changes between this document and a prior version of the same document.
-    /// The changes, when applied to the text of the old document, will produce the text of the current document.
+    /// Get the text changes between this document and a prior version of the same document. The changes, when applied
+    /// to the text of the old document, will produce the text of the current document.
     /// </summary>
-    public Task<IEnumerable<TextChange>> GetTextChangesAsync(Document oldDocument, CancellationToken cancellationToken = default)
+    public async Task<IEnumerable<TextChange>> GetTextChangesAsync(Document oldDocument, CancellationToken cancellationToken = default)
     {
-        return Task.FromResult<IEnumerable<TextChange>>(GetTextChangesSynchronously(oldDocument, cancellationToken));
+        return await GetTextChangesAsync(useAsync: true, oldDocument, cancellationToken).ConfigureAwait(false);
```

Comment:
What is the benefit of awaiting this task here instead of just returning it?

Reply:
one returns an ImmutableArray, the other an IEnumerable.  We can't change the IEnumerable (part of hte public API).  but it means callers to the internal methods don't deal with the confusing, potentially lazy, call signature.